services/service_provider.go

```diff
@@ -26,7 +28,7 @@ var vaultService security.VaultAPI
 // initialized once in the main.go. If it wouldn't work, main would
 // os.Exit(1) and the rest of the application would just stop.
 func StorageService() (store.GaiaStore, error) {
-	if storeService != nil {
+	if storeService != nil && !reflect.ValueOf(storeService).IsNil() {
```

Comment:
So the reflect here is because Go's interface is not really just one value. It's a value and a type. So if the mock is set to nil, the before hand nil check won't be saying that it's not nil. It will be of the previous type which won't equal the nil type, thus a simple `!=` won't work. But we still need it because when the system is starting up it is actually nil. :) 

I love go, but this is just insane. :D Oh and it's totally only needed because of the tests. Otherwise we don't really care, because once the application is running, it's running. We don't delete the db from underneath it randomly.

Reply:
Yeah that is one of the bad parts about Go. I think you solved it quite nicely 🤗 GJ